source/core/ExecutionContext.cpp

```diff
@@ -134,6 +140,17 @@ VIREO_FUNCTION_SIGNATURE1(FPSync, StringRef)
 #endif
     return _NextInstruction();
 }
+
+// generic implementation: debug node
+VIREO_FUNCTION_SIGNATURE1(DebugNode, StringRef)
```

Comment:
Why do you need this additional DebugNode function in ExecutionContext? We already have a function which handles DebugPoint instructions in DebugPoint.cpp. Also we have a DebuggingContext.cpp already setup which has the map for DebugPointState which you can query. I think you should move the extern to DebuggingContext instead and call it from the DebugPoint implementation based on whether [GetDebugPointState](https://github.com/shivaprasad-basavaraj/VireoSDK/blob/191da8fad4e5e94deb3c2b47a2d6600ac41b676a/source/include/DebuggingContext.h#L17) returns true.